Week7/MyFavoriteMovies/app/src/main/java/com/raywenderlich/myfavoritemovies/activities/MainActivity.kt

```diff
@@ -27,15 +28,40 @@ class MainActivity : AppCompatActivity() {
     private val movieRepository by lazy { MovieRepository() }
     private val userRepository by lazy { UserRepository() }
     private val movieAdapter by lazy { MovieAdapter(::movieItemClicked, ::movieItemLongClicked) }
+    private val remoteApi = App.remoteApi
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        getMovieFromApi("Star Wars: Episode I")
+        getMovieFromApi("Star Wars: Episode II")
+        getMovieFromApi("Star Wars: Episode III")
+        getMovieFromApi("Star Wars: Episode IV")
+        getMovieFromApi("Star Wars: Episode V")
+        getMovieFromApi("Star Wars: Episode VI")
+        getMovieFromApi("Star Wars: Episode VII")
+        getMovieFromApi("Star Wars: Episode VIII")
+        getMovieFromApi("Star Wars: Episode IX")
+        fillMovieList()
+
+    }
 
+    private fun getMovieFromApi(movieName: String) {
         lifecycleScope.launch {
-            movieRepository.storeMoviesIfNotEmpty(movies)
+            val result = remoteApi.getMovie(movieName)
+            if (result is Success) {
+                movieRepository.insertMovie(
+                    Movie(
+                        result.data.data.movies[0].idIMDB,
+                        result.data.data.movies[0].releaseDate,
+                        result.data.data.movies[0].title,
+                        result.data.data.movies[0].plot,
+                        result.data.data.movies[0].genres,
+                        result.data.data.movies[0].urlPoster
+                    )
+                )
+            }
         }
-        fillMovieList()
     }
 
     private fun fillMovieList() {
```

Comment:
In `fillMovieList()` you don't have to launch coroutines. As you do that just before you fetch the data.